### PR TITLE
Move the Markdown component to its own package

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- This moves the `<Markdown>` component to its own package.
- Package is called `@astrojs/markdown`. Can still change if another name is preferred.
- Depends on https://github.com/withastro/compiler/pull/472

## Testing

- Moved over as many of the tests as I could.

## Docs

- https://github.com/withastro/docs/pull/1045

### CODE REVIEW

File: packages/markdown/component/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "type": "git",
+    "url": "https://github.com/withastro/astro.git",
+    "directory": "packages/markdown/component"
+  },
+  "bugs": "https://github.com/withastro/astro/issues",
+  "homepage": "https://astro.build",
+  "exports": {
+    ".": {
+      "astro": "./Markdown.astro"
+    }
```
Comment (natemoo-re) [inline packages/markdown/component/package.json:17]: This works, but any reason not to use the normal `".": "./Markdown.astro"` here?

File: packages/markdown/component/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@astrojs/markdown",
+  "version": "0.1.0",
+  "type": "module",
```
Comment (natemoo-re) [inline packages/markdown/component/package.json:4]: Do we need a `types` field as well for TypeScript support in the Language Server?

File: packages/markdown/component/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# `<Markdown />`
```
Comment (natemoo-re) [inline packages/markdown/component/readme.md:1]: Would be nice to align this README with the other integration READMEs.

**Review by natemoo-re — COMMENTED** [review]
RIP! Looks good but had a few questions.

File: packages/markdown/component/package.json (hunk 1 above)
Comment (matthewp) [reply packages/markdown/component/package.json:17]: Just seemed more correct, this package won't work outside of Astro, won't work in regular Node.js ESM, and the error message seemed more appropriate if you tried.

File: packages/markdown/component/package.json (hunk 2 above)
Comment (matthewp) [reply packages/markdown/component/package.json:4]: Yeah, probably, good call.

Comment (matthewp) [reply packages/markdown/component/package.json:4]: Actually, just adding a `main` field should work.

File: packages/markdown/component/readme.md (hunk 3 above)
Comment (sarah11918) [reply packages/markdown/component/readme.md]: The MDX integration README in particular is probably a good comp to use (no configuration etc): https://github.com/withastro/astro/blob/main/packages/integrations/mdx/README.md

**Comment by delucis** [comment]
Not a showstopper, but do we think that `@astrojs/markdown` is too juicy a package name? Might attract folks when we really want to put them off using this? Maybe even something as explicit as `@astrojs/markdown-component` or `@astrojs/static-markdown-component` to make it seem less enticing.

**Review by tony-sull — APPROVED** [review]
lgtm!

**Pull request merged**